### PR TITLE
Feature/add dwelling

### PR DESCRIPTION
**Pull request by xspace-solution**

### CODE REVIEW

File: src/modules/assetus/briefs4assetus/asset_brief.go [hunk 1]
```diff
@@ -103,12 +104,13 @@ func (v *AssetBrief) Validate() error {
 		return validation.NewErrBadRecordFieldValue("category", "unknown asset category: "+string(v.Category))
 	}
 
-	if strings.TrimSpace(v.Make) == "" {
-		return validation.NewErrRecordIsMissingRequiredField("make")
-	}
-	if strings.TrimSpace(v.Model) == "" {
-		return validation.NewErrRecordIsMissingRequiredField("model")
-	}
+	// if strings.TrimSpace(v.Make) == "" {
```
Comment (trakhimenok) [inline src/modules/assetus/briefs4assetus/asset_brief.go:107]: This need to be inside IF block for vehicle only

File: src/modules/assetus/const4assetus/asset_category.go [hunk 2]
```diff
@@ -7,8 +7,8 @@ const (
 	// AssetCategoryVehicle = "vehicle
 	AssetCategoryVehicle AssetCategory = "vehicle"
 
-	// AssetCategoryRealEstate = "real_estate"
-	AssetCategoryRealEstate AssetCategory = "real_estate"
+	// AssetCategoryRealEstate = "dwelling"
+	AssetCategoryRealEstate AssetCategory = "dwelling"
```
Comment (trakhimenok) [inline src/modules/assetus/const4assetus/asset_category.go:11]: Please rename the const to dwelling for consistency

File: src/modules/assetus/briefs4assetus/asset_brief.go (hunk 1 above)
Comment (xspace-solution) [reply src/modules/assetus/briefs4assetus/asset_brief.go:107]:
```
case const4assetus.AssetCategoryVehicle:
		if strings.TrimSpace(v.Make) == "" {
			return validation.NewErrRecordIsMissingRequiredField("make")
		}
		if strings.TrimSpace(v.Model) == "" {
			return validation.NewErrRecordIsMissingRequiredField("model")
		}
		if err := checkType(const4assetus.AssetVehicleTypes); err != nil {
			return err
		}
```
I moved Make and Model validation here.

File: src/modules/assetus/models4assetus/asset_dwelling.go [hunk 3]
```diff
@@ -1,22 +1,22 @@
...
 
 type AssetDwellingExtra struct {
 	AssetExtraBase
-	BedRooms  int `json:"bedRooms,omitempty" firestore:"bedRooms,omitempty"`
-	BathRooms int `json:"bathRooms,omitempty" firestore:"bathRooms,omitempty"`
+	// BedRooms  int `json:"bedRooms,omitempty" firestore:"bedRooms,omitempty"`
+	// BathRooms int `json:"bathRooms,omitempty" firestore:"bathRooms,omitempty"`
 }
 
 func (v AssetDwellingExtra) Validate() error {
```
Comment (trakhimenok) [inline src/modules/assetus/models4assetus/asset_dwelling.go:11]: Here we define extra fields specific to asset category

File: src/modules/assetus/models4assetus/asset_dwelling.go [hunk 4]
```diff
@@ -1,22 +1,22 @@
...
 	if err := v.AssetExtraBase.Validate(); err != nil {
 		return err
 	}
-	if v.BedRooms < 0 {
-		return validation.NewErrBadRecordFieldValue("bedRooms", "negative value")
-	}
-	if v.BathRooms < 0 {
-		return validation.NewErrBadRecordFieldValue("bathRooms", "negative value")
-	}
+	// if v.BedRooms < 0 {
```
Comment (trakhimenok) [inline src/modules/assetus/models4assetus/asset_dwelling.go:15]: This should not be commented out and an `AreaSqM` to be added here as well

**Pull request merged**